From ef54a6f4a81da8ab653d352bfbd3b2521ce9eb7c Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 29 Aug 2022 15:55:32 -0700
Subject: [PATCH] iscsiuio: Use pthread_t for INVALID_THREAD

pthread_t is opaque, therefore avoid compiler errors on musl when
compiling since pthread_t is not a plain old data type, like glibc

Upstream-Status: Submitted [https://github.com/open-iscsi/open-iscsi/pull/363]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 iscsiuio/src/unix/options.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iscsiuio/src/unix/options.h b/iscsiuio/src/unix/options.h
index 63b8635..873a98a 100644
--- a/iscsiuio/src/unix/options.h
+++ b/iscsiuio/src/unix/options.h
@@ -86,7 +86,7 @@
 #define DEBUG_ON	0x2
 
 #define INVALID_FD	-1
-#define INVALID_THREAD	-1
+#define INVALID_THREAD	(pthread_t)-1
 #define INVALID_HOST_NO	-1
 
 struct options {
